setup/security/authentication/ssl.md

```diff
@@ -29,66 +29,178 @@ use an external CA, to minimize browser configuration, but it's not necessary
 Use the steps below to create a certificate authority. (If you're using an
 external CA, skip to the next section.)
 
-1. Create the CA key.
-   ```
-   openssl genrsa -des3 -out ca.key 4096
-   ```
+It will produce the following items:
 
-1. Self-sign the CA certificate.
-   ```
-   openssl req -new -x509 -days 365 -key ca.key -out ca.crt
-   ```
+* `ca.key`: a `pem`-formatted private key, which will have a pass phrase.
+* `ca.crt`: a `pem`-formatted certificate, which (with the private key) acts as
+a self-signed Certificate Authority.
 
-## 2. Create the server certificate
+1. Create the CA key.  This will prompt for a pass phrase to encrypt the key.
 
-Use the self-signed CA cert you created above to sign the server certificate.
+    ```bash
+    openssl genrsa -des3 -out ca.key 4096
 
-1. Create the server key. Keep this file safe!
-```
-openssl genrsa -des3 -out server.key 4096
-```
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passout pass:$CA_KEY_PASSWORD`
```

Comment:
These optional flags seem a bit verbose, could you remove them and add a section for pass phrase flags with openssl?